src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/Construction/Solution.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    /// <summary>
+    /// Defines a <c>.sln</c> file to be created when using <see cref="ProjectLayoutTestBase"/>.
+    /// </summary>
+    public sealed class Solution : IEnumerable
+    {
```

Comment:
Does this need to implement `IEnumerable` if the `IEnumerable.GetEnumerator()` implementation just throws an exception?

Reply:
For some definition of "need". But yes, it's intentional. If you want to use C# collection initialisers, you need to implement `IEnumerable` and have at least one `void Add(...)` method.